pkg/pipeline/extract/timebased/filters.go

```diff
@@ -149,10 +150,21 @@ func (fs *FilterStruct) CreateGenericMap() []config.GenericMap {
 			"index_key":        fs.Rule.IndexKey,
 			"operation":        fs.Rule.OperationType,
 			"operation_key":    fs.Rule.OperationKey,
-			"key":              result.key,
+			"key":              result.values,
 			"operation_result": result.operationResult,
 		}
-		t[fs.Rule.IndexKey] = result.key
+
+		// append operation key and result as key / value
+		t[fs.Rule.OperationKey] = result.operationResult
+
+		// append index key / value pairs
+		values := strings.Split(result.values, ",")
+		if len(fs.Rule.IndexKeys) == len(values) {
+			for i, k := range fs.Rule.IndexKeys {
+				t[k] = values[i]
+			}
+		}
+
```

Comment:
Once you've added in the individual fields, I don't see the need to keep the `key`, `operation_key`, and `operation_result` fields.

Reply:
Yes exactly; done in https://github.com/netobserv/flowlogs-pipeline/pull/450/commits/25842d7f598b04d13e02897c5fba9a7d7d0ddb4f